### PR TITLE
Add config gem so we can use Settings.yml etc.

### LINKED ISSUE #21: PCC: use settings.yml instead of hardcoding
we have a branch (`show-action-naomi`) with a clear need for some settings so this can be set up off that branch (or master, if PR #25 is merged)

e.g.  need config/settings.yml  with values to be used by `config/application.rb`

We want to do the same sort of configuration approach we use in most rails apps, with the `config` gem and `config/settings.yml` with defaults and `config/settings/[env].yml` also with defaults, with no sensitive info in `config/settings/production.yml`.   `dlme` project might be a good example.

This will set us up for #4, where we put VM specific or sensitive information into shared_configs and hook it up so rails app finds it as if it was in `config/settings/[env].yml`

### PR DESCRIPTION
**Pull request by tingulfsen**
Fixes #21

### CODE REVIEW

File: config/initializers/config.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  # Ability to process variables names:
+  #   * nil  - no change
+  #   * :downcase - convert to lower case
+  #
+  # config.env_converter = :downcase
+
+  # Parse numeric values as integers instead of strings.
+  #
+  # config.env_parse_values = true
```
Comment (jmartin-sul) [inline config/initializers/config.rb]: on the one hand, this feels like it runs afoul of what we talked about this morning where we don't include commented out boilerplate.  on the other hand, i'd have been inclined to leave it also.  i don't think i have an opinion?  what do others think?

**Comment by ndushay** [comment]
In the end, I don't think we check in `development.yml` either after we have shared_configs, but we do need an example  .yml and README instructions on how to turn example.yml (or test.yml) into development.yml for local development.

File: config/initializers/config.rb (hunk 1 above)
Comment (ndushay) [reply config/initializers/config.rb]: given that we actually do use configs a lot and I have never noticed all this boilerplate or mucked about with these settings, I would be comfortable removing all this boilerplate.  But I don't have strong feels and we could leave it, too.

**Comment by tingulfsen** [comment]
I think I probably addressed the concerns mentioned above, rebased on master. I left development.yml and test.yml in there for the moment. We can remove those too if you like, but maybe it's easier to leave them since we know we will need them but don't have any offending data yet.

**Comment by ndushay** [comment]
👍  to merge.  @jmartin-sul  I'll leave it to you ...

**Pull request merged**